File: leetcode/python/2419.longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        maxn = max(nums)
        maxLen = 1
        count = 0
        for n in nums:
            if n == maxn:
                count += 1
            else:
                count = 0
            maxLen = max(count, maxLen)
        return maxLen

File: leetcode/rust-leetcode/src/2419.longest-subarray-with-maximum-bitwise-and.rs
impl Solution {
    pub fn longest_subarray(nums: Vec<i32>) -> i32 {
        let maxn = *nums.iter().max().unwrap();
        let mut ans = 1;
        let mut count = 0;
        for n in nums {
            if n == maxn {
                count += 1;
                ans = ans.max(count);
            } else {
                count = 0;
            }
        }
        ans
    }
}

